pkg/apis/events/v1/conversion.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	v1 "k8s.io/api/events/v1"
+	conversion "k8s.io/apimachinery/pkg/conversion"
+	k8s_api "k8s.io/kubernetes/pkg/apis/core"
+	k8s_api_v1 "k8s.io/kubernetes/pkg/apis/core/v1"
+)
+
+func Convert_v1_Event_To_core_Event(in *v1.Event, out *k8s_api.Event, s conversion.Scope) error {
+	if err := autoConvert_v1_Event_To_core_Event(in, out, s); err != nil {
+		return err
+	}
+	if err := k8s_api_v1.Convert_v1_ObjectReference_To_core_ObjectReference(&in.Regarding, &out.InvolvedObject, s); err != nil {
+		return err
+	}
+	if in.DeprecatedSource.Component == in.ReportingController {
+		if err := k8s_api_v1.Convert_v1_EventSource_To_core_EventSource(&in.DeprecatedSource, &out.Source, s); err != nil {
+			return err
+		}
+	} else {
+		out.Source.Component = in.ReportingController
+	}
```

Comment:
I'm not sure I understand this... this seems to be discarding in.DeprecatedSource under some circumstances... is there a reason not to match v1beta1 conversion:

```
	if err := k8s_api_v1.Convert_v1_EventSource_To_core_EventSource(&in.DeprecatedSource, &out.Source, s); err != nil {
		return err
	}
```

Reply:
I believe the initial intent was to address this TODO: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/events/event_recorder.go#L88, but yes this does have a problem - it will fail RoundTripTypes test. So we decided to leave it as is.